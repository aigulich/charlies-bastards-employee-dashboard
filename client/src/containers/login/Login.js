import React from 'react'
import {useForm} from 'react-hook-form'
import loginHelper from "../../services/loginHelper"
import { connect,useDispatch} from  'react-redux';
import {createSession} from "../../reducers/actions/index"
function Login(){
const [message,setMessage] = React.useState("")
 const {register,errors, handleSubmit} = useForm()
  const dispatch = useDispatch()
 /**
  * @function onSumit
  * @param {string} data -Values passed in the input 
  * @param {*} e - event 
  */
 
 const onSubmit = (data, e) => {
     e.preventDefault()
     loginHelper(data.email, data.password)
     .then(res=>
          dispatch(createSession(
              res.data.message,
              res.data.token,
              res.data.firstname,
              res.data.lastname))  
        )
       .catch(err=>  setMessage(`${err.response.data.message.message}`) )

  };
    return(
        <div data-test="login-component">
            <h1 data-test="h1-login-component">{message}</h1>
            <form data-test="form-component" onSubmit={handleSubmit(onSubmit)}>
                <label htmlFor="email">E-mail</label>
                <input
                data-test="input-form"
                 type="text"
                  name='email'
                  ref={register({required:true})}/>
                  {errors.email && "This field is required"}
                <label htmlFor="password">Password</label>
                 <input 
                 data-test="input-form"
                 type="password"
                  name='password'
                  ref={register({required:true})}/>
                  {errors.password && "This field is required"}
                 <button  data-test="submit-button" type="submit"> Login  </button>
            </form>

        </div>
    )

}
export default Login