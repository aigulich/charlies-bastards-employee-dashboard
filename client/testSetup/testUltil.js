import rootReducer from "../src/reducers"
import {createStore} from 'redux'

/**
 * @function storeFactory
 * @param {object} initialState  -initial State for the store
 * @returns {Store}- Redux Store
 */

 export const storeFactory=(initialState)=>{
   return  createStore(rootReducer,initialState)     
 }

 


 /**
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper to search within
 * @param {ShallowWrapper}
 * @param {string} val - Value of data-test atribute for search 
 */
export const findByTestAttr = (wrapper, val)=>{
        return  wrapper.find(`[data-test="${val}"]`)
       
}



export const initialState ={
        state:[],
        reducer:{}

}

/**
 * @function expectTruthy
 * @param {function getAllByTestId("testid")||getByTestId("testid") {
       return elements in the DOM 
 }} receiveObj 
 */

export const expectTruthy = (receiveObj)=>{
        expectTruthy(receiveObj).toBeTruthy() && expect(receiveObj).not.toBeTruthy()
}