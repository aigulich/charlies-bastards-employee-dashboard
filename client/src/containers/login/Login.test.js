import React from "react";
import Login  from "./Login";
import { Provider} from 'react-redux';
import Enzyme, { mount, ShallowWrapper,shallow } from "enzyme";
import EnzymeAdapter from "enzyme-adapter-react-16";
import {findByTestAttr,initialState,storeFactory, expectTruthy} from "../../../testSetup/testUltil"
import configureStore from 'redux-mock-store'
import { render, fireEvent} from "@testing-library/react";
import { BrowserRouter as Router}from "react-router-dom"
  Enzyme.configure({ adapter: new EnzymeAdapter() });

/* describe('test if component renders without error', ()=>{
  const {getByTestId, getAllByTestId} =render(<Login/>)
  test('display all forms elements',()=>{
    expectTruthy(getAllByTestId('test-label'))
  })
}) */



/* test('renders without error', ()=>{
  
    const wrapper = shallow( <Login/>
    )
expect(wrapper).toMatchSnapshot()
    
      }) */
      


    test('login component',  ()=>{
      const historyMock = { push: jest.fn() };
      const mockStore = configureStore()
        const store = mockStore(initialState)
        const {getByTestId} = render(
          <Router>
          <Provider store={store} >
                <Login history={historyMock}/>
             </Provider>
             </Router>
        )
        console.log(getByTestId("submit-button"))  
             /*  const submitButton= findByTestAttr(wrapper,"submit-button")
        submitButton.simulate('click')
        expect(historyMock.push).toBeCalled() */
        });
      

 

/* const setup = (initialState={}, props={}) =>{
const store = storeFactory(initialState)
    const wrapper = shallow(<Provider store={store}><Login/></Provider>).dive().dive()
  console.log(wrapper.debug())}

  setup()
  */


 /* test('login component',  ()=>{
     const componentLogin= findByTestAttr(wrapper,"login-component")
    expect(componentLogin.length).toBe(1) 
  });*/

/* 
describe ('renders without an error ', () => {
    test('login component',  ()=>{
        const componentLogin= findByTestAttr(wrapper,"login-component")
        expect(componentLogin.length).toBe(1)
      });
      test('form', () => {
        const componentForm= findByTestAttr(wrapper,"form-component")
        expect(componentForm.length).toBe(1)
      })
      test(' input', ()=>{
        const InputForm= findByTestAttr(wrapper,"input-form")
        expect(InputForm.length).toBe(2)
        })
      test(' submit button ', ()=>{
        const submitButton= findByTestAttr(wrapper,"submit-button")
        expect(submitButton.length).toBe(1)
        }) 
  
})
 */

