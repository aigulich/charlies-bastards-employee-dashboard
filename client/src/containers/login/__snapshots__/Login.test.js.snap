// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login component 1`] = `ReactWrapper {}`;
